cmake/third_party/gcp_sdk.cmake

```diff
@@ -0,0 +1,75 @@
+cmake_minimum_required(VERSION 3.10)
```

Comment:
you can remove this since it is documented that we need 3.10+ anyways

Reply:
Do we have other `cmake_minimum_required` checks present? If not I'd prefer to keep this as it's clearer to have this line print an error rather than requiring a user to read through the docs.

Reply:
Yep we have this check present in the root cmakelist.txt file.

Reply:
Don't we need cmake 3.13+? Has that issue been sorted?

Reply:
That is for the azure SDK extension, looking at the documentation 3.10+ is enough for GCP

Reply:
Got it, thanks!